plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/layout/provider/BorderItemAwareLayoutProvider.java

```diff
@@ -870,11 +878,13 @@ private Command layoutBorderItems(final GraphicalEditPart graphicalEditPart, Arr
             }
 
         }
-        for (Object editPart : graphicalEditPart.getChildren()) {
-            if (editPart instanceof GraphicalEditPart) {
-                final Command layoutBorderItems = layoutBorderItems((GraphicalEditPart) editPart, elementsToKeepFixed);
-                if (layoutBorderItems != null && layoutBorderItems.canExecute()) {
-                    result.add(layoutBorderItems);
+        if (recursive) {
```

Comment:
It seems that the method is not recursive but just applying on children editPart. Am I right? And, if yes, is it what we want

Reply:
No, this method is recursive, see the call line [line 875/884](https://github.com/eclipse-sirius/sirius-desktop/pull/142/files#diff-cc5ce8813d519c48ba2840162ddf69f330ac69411540cb46440b2832474a2bc7R884). We want recursive call when the parameter recursive is `true` and we set recursive true when we arrange port of the whole diagram (because in this case, the selection is the diagram alone) and `false` for arrange port of selection.